ambry-network/src/test/java/com.github.ambry.network/NetworkClientTest.java

```diff
@@ -155,7 +155,7 @@ public void testBasicSendAndPoll() {
     int responseCount = 0;
 
     do {
-      responseInfoList = networkClient.sendAndPoll(requestInfoList, POLL_TIMEOUT_MS);
+      responseInfoList = networkClient.sendAndPoll(requestInfoList, Collections.emptyList(), POLL_TIMEOUT_MS);
```

Comment:
We should write a real test cases for dropping connections.

Reply:
I wrote a new test case in NetworkClientTest that should cover all of the new NetworkClient functionality